controllers/auth.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+const { validate } = require('indicative').validator;
+const { AuthorizationError } = require('errors');
+
+const FreedomHelper = require('helpers/freedom_helper');
+const User = require('models/User');
+
+const jwt = require('lib/jwt');
+const responses = require('lib/responses');
+
+module.exports = {
+    login,
+    logout
+};
+
+async function login (req) {
+    const data = req.body;
+
+    const rules = {
+        code: 'required|string',
+        redirect_uri: 'required|url',
+        grant_type: 'required|string'
+    };
+
+    const validated_data = await validate(data, rules);
+    let oauth_response = {};
+
+    try {
+        oauth_response = await FreedomHelper.get_oauth_access_token(validated_data);
+    }
+    catch (err) {
+        throw new AuthorizationError(responses.OAUTH_ERROR);
+    }
+
+    if (!oauth_response.access_token) {
+        throw new AuthorizationError(responses.OAUTH_ERROR);
+    }
+
+    const user_info = await FreedomHelper.get_user_information({
+        access_token: oauth_response.access_token
+    });
+
+    const user = new User(user_info);
```

Comment:
Check if user_info exist.

Reply:
Added an `if` statement to check first if `user_info` exists.